kivy/core/text/markup.py

```diff
@@ -627,6 +627,14 @@ def p_restricted(line, uw, c):
             return lw + 2 * xpad, lh + 2 * ypad, [LayoutLine(0, 0,
             lw, lh, 1, 0, line)]
 
+        elps_opts = copy(old_opts)
+        if 'ellipsis_options' in old_opts:
```

Comment:
This whole block can be written as `elps_opts.update(old_opts.get('ellipsis_options', {}))`. It's up to you if you want to change it, but either way feel free to merge.